marlowe-dashboard-client/src/API/Marlowe/Run/Wallet/CentralizedTestnet.purs

```diff
@@ -1,55 +1,130 @@
 module API.Marlowe.Run.Wallet.CentralizedTestnet
-  ( restoreWallet
+  ( createWallet
+  , clientServerError
+  , ClientServerErrorRow
+  , CreateWalletError
+  , CreateWalletResponse
+  , mkRestoreWalletError
+  , ServerErrorRow
+  , restoreWallet
   , RestoreWalletOptions
-  , RestoreError(..)
+  , RestoreWalletError(..)
   ) where
 
 import Prologue
 
 import API.Request (doPostRequestWith)
-import Cardano.Wallet.Mock.Types (WalletInfo)
-import Control.Monad.Except (runExceptT)
+import Bridge (toFront)
+import Cardano.Wallet.Mock.Types (WalletInfo) as Mock.Types
+import Component.Contacts.Types (WalletInfo)
+import Control.Error.Util (exceptNoteM)
+import Control.Monad.Except (except, runExceptT, withExceptT)
 import Data.Argonaut (encodeJson)
 import Data.Argonaut.Decode.Aeson as D
+import Data.Bifunctor (bimap, lmap)
+import Data.MnemonicPhrase (MnemonicPhrase, MnenonicPhraseErrorRow)
+import Data.MnemonicPhrase (fromStrings) as MnemonicPhrase
+import Data.MnemonicPhrase (injErr) as MnenonicPhrase
+import Data.Passpharse (Passphrase)
+import Data.Passpharse (toString) as Passphrase
+import Data.Variant (Variant)
+import Data.Variant (inj) as Variant
+import Data.Variant.Generic
+  ( class Constructors
+  , mkConstructors
+  , mkConstructors'
+  )
+import Data.WalletNickname (WalletNickname)
+import Data.WalletNickname (toString) as WalletNickname
 import Effect.Aff.Class (class MonadAff)
 import Marlowe.Run.Wallet.V1.CentralizedTestnet.Types
-  ( RestoreError(..)
+  ( CreatePostData(..)
+  , RestoreError(..)
   , RestorePostData(..)
   ) as BE
+import Marlowe.Run.Wallet.V1.CentralizedTestnet.Types (CreateResponse(..))
 import Servant.PureScript (AjaxError)
+import Type.Prelude (Proxy(..))
+import Type.Row (type (+))
 
 type RestoreWalletOptions
   =
-  { walletName :: String, mnemonicPhrase :: Array String, passphrase :: String }
+  { walletName :: String
+  , mnemonicPhrase :: Array String
+  , passphrase :: Passphrase
+  }
+
+type ClientServerErrorRow r = (clientServerError :: AjaxError | r)
```

Comment:
Why do you add the Row suffix?

Reply:
So I can "add" rows together:

```purescript
type RestoreWalletError = Variant  ( ClientServerErrorRow + ServerErrorRow BE.RestoreError + (invalidMnemonic :: Unit))
```

It is also useful when we have polyfmorphic context. Leaving row types open is in general a good practice because they are "reusable" and we can always just close them when we need to: `Variant (ClientServerErrorRow ())`